decanter_ai_sdk/non_blocking_client.py

```diff
@@ -0,0 +1,641 @@
+                Feature names that are not going to be used during experiment.
+            evaluator (Union[`~decanter_ai_sdk.enums.evaluators.ClassificationMetric`, `~decanter_ai_sdk.enums.evaluators.RegressionMetric`])
+                Evaluator used as stopping metric.
+            holdout_table_id (str)
+                Id of the table used to perform holdout.
+            algos (Union[List[`~decanter_ai_sdk.enums.algorithms.IIDAlgorithms`], List[`~decanter_ai_sdk.enums.algorithms.TSAlgorithms`]])
+                Algorithms used for experiment.
+            max_model (int)
+                Limit for the number of models to train for this experiment.
+            tolerance (int)
+                Larger error tolerance will let the training stop earlier. Smaller error tolerance usually generates more accurate models but takes more time. (1~10)
+            nfold (int)
+                Amount of folds in experiment. (2~10) for autoML. (1~10) for autoTSF.
+            stacked_ensemble (boolean)
+                If stacked ensemble models will be trained.
+            validation_percentage (int)
+                Validation percentage of experiment. (5~20)
+            seed (int)
+                Random Seed of experiment. (1 ~ 65535)
+            timeseries_value (List[Dict[str, Any]])
+                Folds for time series cross validation (train, window, test, holdout_timeseries, cv, holdout_Percentage, split_By, lag).
+                \tExample:
+                    {
+                        'train': {
+                            'start': '2011-03-18T00:00:00',
+                            'end': '2012-04-21T00:00:00'
+                        },
+                        'window': 40,
+                        'test': 20,
+                        'holdout_timeseries': {
+                            'start': '2012-04-21T00:00:00',
+                            'end': '2020-03-28T00:00:00'
+                        },
+                        'cv': [
+                            {
+                                'train': {
+                                    'start': '2009-10-25T00:00:00',
+                                    'end': '2010-11-29T00:00:00'
+                                },
+                                'test': {
+                                    'start': '2010-11-29T00:00:00',
+                                    'end': '2011-06-17T00:00:00'
+                                }
+                            }
+                        ],
+                        'holdoutPercentage': 10,
+                        'splitBy': 'MonitorDateTime',
+                        'lag': 0
+                    }
+            holdout_percentage (int)
+                Holdout percentage for experiment.
+
+        Returns:
+        ----------
+            (`~decanter_ai_sdk.web_api.experiment.Experiment`)
+                Experiment id.
+        """
+
+        data_column_info = self.api.get_table_info(table_id=experiment_table_id)
```

Comment:
you want to factor out all this duplicate code to helper functions that can be shared between the two clients. There should never be this much duplicate code. It makes it more difficult when it needs to be modified in the future

Reply:
Roger that!